contracts/sentinel/PCVSentinel.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "../refs/CoreRef.sol";
+import "../pcv/IPCVDeposit.sol";
+import "./IPCVSentinel.sol";
+import "./IGuard.sol";
+
+/**
+ * @title PCV Sentinel
+ * @dev the PCV Sentinel should be granted the role Guardian
+ * @notice The PCV Sentinel is a automated extension of the Guardian.
+ * The Guardian can add Guards to the PCV Sentinel. Guards run checks
+ * and provide addresses and calldata for the Sentinel to run, if needed.
+ */
+contract PCVSentinel is IPCVSentinel, CoreRef {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // The set of all guards
+    EnumerableSet.AddressSet private guards;
+
+    constructor(address _core) CoreRef(_core) {
+        _setContractAdminRole(keccak256("PCV_SENTINEL_ADMIN_ROLE"));
+    }
+
+    // ---------- Read-Only API ----------
+
+    /**
+     * @notice returns whether or not the given address is a guard
+     * @param guard the address to check
+     */
+    function isGuard(address guard) external view override returns (bool) {
+        return guards.contains(guard);
+    }
+
+    /**
+     * @notice returns a list of all guards
+     */
+    function allGuards() external view override returns (address[] memory all) {
+        all = new address[](guards.length());
+
+        for (uint256 i = 0; i < guards.length(); i++) {
+            all[i] = guards.at(i);
+        }
+
+        return all;
+    }
+
+    // ---------- Governor-or-Admin-Or-Guardian-Only State-Changing API ----------
+
+    /**
+     * @notice adds a guard contract to the PCV Sentinel
+     * @param guard the guard-contract to add
+     */
+    function knight(address guard)
+        external
+        override
+        isGovernorOrGuardianOrAdmin
```

Comment:
Should a guardian be able to add a guard? This is essentially saying that guardian can add a contract into the system that can construct calldata that will be executed by a contract with PCV Controller role. It feels like this gives the guardians too much power.

Reply:
@ElliotFriedman no that's not the case, where are you getting the idea that this is a PCVController still?

(It was, but shouldn't be anymore)

Reply:
Just confirmed, yeah, this will use the PCVGuardian to pull funds to safety :)

Reply:
Closing due to lack of further response, feel free to re-open @ElliotFriedman if you have further concerns.